src/Elastic.Markdown/DocumentationGenerator.cs

```diff
@@ -231,6 +232,29 @@ private async Task ExtractEmbeddedStaticResources(Cancel ctx)
 		}
 	}
 
+	[GeneratedRegex(@"^[a-z0-9\s\-_\.\/\\]*[a-z0-9_\-]\.([a-z]+)$")]
+	private static partial Regex FilePathRegex();
+
+	[GeneratedRegex(@"^[a-z0-9_][a-z0-9_\-\s\.]*?\.([a-z]+)$")]
+	private static partial Regex FileNameRegex();
+
+	public static bool IsValidFileName(string strToCheck) =>
+		strToCheck switch
+		{
+			//prior art
+			_ when strToCheck.StartsWith("release-notes/elastic-agent/_snippets/") => true,
+			_ when strToCheck.StartsWith("reference/query-languages/esql/_snippets/") => true,
+			_ when strToCheck.EndsWith(".svg") => true,
+			_ when strToCheck.EndsWith(".gif") => true,
+			_ when strToCheck.EndsWith(".png") => true,
+			_ when strToCheck.EndsWith(".png") => true,
+			"reference/security/prebuilt-rules/audit_policies/windows/README.md" => true,
+			"extend/integrations/developer-workflow-fleet-UI.md" => true,
```

Comment:
Not sure if directly related to this change, but docs builds in the integrations repository are failing since around the time it was merged. With the following error:
```
Error: File name extend/developer-workflow-fleet-UI.md is not valid needs to be lowercase and contain only alphanumeric characters, spaces, dashes, dots and underscores
```

First failed build is this one: https://github.com/elastic/integrations/actions/runs/20336771987/job/58425304911

Should this path be modified like this?

```suggestion
			"extend/developer-workflow-fleet-UI.md" => true,
```

cc @elastic/ecosystem @elastic/docs-engineering